src/main/java/org/jasig/portlet/announcements/mvc/servlet/RssFeedController.java

```diff
@@ -73,6 +75,74 @@
             "%s/%s/p/%s?pP_announcementId=%d&pP_action=" + AnnouncementsViewController.ACTION_DISPLAY_FULL_ANNOUNCEMENT;
 
     private static final String PATH_ATTRIBUTE = "path";
+    private static final String FILENAME_ATTRIBUTE = "filename";
+
+    /**
+     * A Content-Type is required for RSS &lt;enclosure&gt; elements.
+     */
+    private enum TypeMapping {
+
+        GIF("image/gif"){
```

Comment:
Would it be possible to leverage https://dzone.com/articles/determining-file-types-java ?

Reply:
This in particular looks promising

https://docs.oracle.com/javase/8/docs/api/javax/activation/MimetypesFileTypeMap.html#getContentType-java.lang.String-

```java
   final MimetypesFileTypeMap fileTypeMap = new MimetypesFileTypeMap();
   return fileTypeMap.getContentType(fileName);
```